pkg/ioutils/fswriters.go

```diff
@@ -12,7 +13,8 @@ type AtomicFileWriterOptions struct {
 	// If NoSync is not specified, the file is synced to the
 	// storage after it has been written and before it is moved to
 	// the specified path.
-	NoSync bool
+	NoSync  bool
+	ModTime time.Time
```

Comment:
Returning a value in “options” seems very surprising to me. At the very least, it should be clearly documented.

Going by the “`…Options` is input to the operation” logic, a `WriteModTimeOnSuccess *time.Time` field would make at bit more sense … and, OTOH, be more cumbersome to use.

Ultimately it would be cleanest to add a new function that returns that value (or a struct containing that value, for future expansion?)

Reply:
I agree that this is somewhat weird, but the primary driver of this particular API was to avoid changing the API and also not needlessly complicating the full API. It is weird, I agree, even ugly. But it is pretty effective, and unlikely to bother anyone. The alternative would be to create yet another "even more options" alternative of the function and have everything chain to that. That is clearly aesthetically better, but increases the amount of api, code and indirect calls at runtime.